api/src/services/xml_generation/transformers/attachment_transformer.py

```diff
@@ -0,0 +1,316 @@
+"""Attachment transformer for XML generation."""
+
+from typing import Any
+
+from lxml import etree as lxml_etree
+
+from ..models.attachment import HASH_ALGORITHM, AttachmentFile, AttachmentGroup
+
+
+class AttachmentTransformer:
+    """Transformer for handling attachment data in XML generation."""
+
+    def __init__(
+        self, attachment_namespace: str = "http://apply.grants.gov/system/Attachments-V1.0"
+    ):
+        """Initialize the attachment transformer.
+
+        Args:
+            attachment_namespace: The XML namespace for attachments
+        """
+        self.attachment_namespace = attachment_namespace
+
+    def add_attachment_elements(
+        self, parent: lxml_etree._Element, data: dict[str, Any], nsmap: dict[str, str]
+    ) -> None:
+        """Add attachment elements to the parent XML element.
+
+        Args:
+            parent: Parent XML element
+            data: Data dictionary containing attachment information
+            nsmap: Namespace map for XML generation
+        """
+        # Handle single attachment fields
+        single_attachment_fields = [
+            ("AreasAffected", "areas_affected"),
+            ("AdditionalCongressionalDistricts", "additional_congressional_districts"),
+            ("DebtExplanation", "debt_explanation"),
+        ]
```

Comment:
Isn't whether a field is a single/multi-attachment in the configuration now? I think between the attachment file and this one, there is a bit of duplicated / extra code.